src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -855,6 +873,36 @@ public int LastIndexOf(T item, int index, int count)
             return SegmentedArray.LastIndexOf(_items, item, index, count);
         }
 
+        public int LastIndexOf(T item, int index, int count, IEqualityComparer<T>? comparer)
+        {
+            if ((Count != 0) && (index < 0))
+            {
+                ThrowHelper.ThrowIndexArgumentOutOfRange_NeedNonNegNumException();
+            }
+
+            if ((Count != 0) && (count < 0))
```

Comment:
```suggestion
            if (Count != 0 && count < 0)
```

Reply:
It could go either way. I probably would have omitted then for brand new code, but I'll keep it for now since it matches the above method.

Reply:
➡️ Going to leave these for now.